src/crypto.ts

```diff
@@ -0,0 +1,69 @@
+import blake from "blakejs";
+import { RegistryValue } from "./registry";
+import { FileID } from "./skydb";
+
+// NewHash returns a blake2b 256bit hasher.
+function NewHash() {
+  return blake.blake2bInit(32, null);
+}
+
+// HashAll takes all given arguments and hashes them.
+export function HashAll(...args: any[]): Uint8Array {
+  const h = NewHash();
+  for (let i = 0; i < args.length; i++) {
+    blake.blake2bUpdate(h, args[i]);
+  }
+  return blake.blake2bFinal(h);
+}
+
+// HashRegistryValue hashes the given registry value and returns it as a hex
+// encoded string
+export function HashRegistryValue(registryValue: RegistryValue): string {
+  return Buffer.from(
```

Comment:
If you're using Buffer then you need the `buffer` library. It is already a dependency on my branch.